homeassistant/components/mqtt/fan.py

```diff
@@ -691,7 +722,7 @@ async def async_set_speed(self, speed: str) -> None:
 
             if self._optimistic_speed and speed_payload:
                 self._speed = speed
-            self.async_write_ha_state()
+                self.async_write_ha_state()
```

Comment:
Is this an unrelated bugfix to avoid unnecessary state machine updates?

Reply:
It is an unrelated bugfix, I made a note in the description of the PR to indicate this fix.